presto-main/src/main/java/com/facebook/presto/cost/FilterStatsCalculator.java

```diff
@@ -160,11 +160,86 @@ protected PlanNodeStatsEstimate visitNotExpression(NotExpression node, Void cont
             if (node.getValue() instanceof IsNullPredicate) {
                 return process(new IsNotNullPredicate(((IsNullPredicate) node.getValue()).getValue()));
             }
-            PlanNodeStatsEstimate sourceStats = process(node.getValue());
-            if (sourceStats.isOutputRowCountUnknown()) {
+            PlanNodeStatsEstimate source = process(node.getValue());
```

Comment:
I don't like the name source here. To me it means the same thing as input, which is confusing. maybe 'negatedFilter'?

Reply:
Aggree. Renamed `source` to `negatedFilter` everywhere in this method